hyperspy/component.py

```diff
@@ -180,16 +183,10 @@ def __len__(self):
         return self._number_of_elements
 
     def connect(self, f):
-        if f not in self.connected_functions:
```

Comment:
Probably we should deprecate this method for removal in 0.10 and avoid using it internally.